---
title: "Read Homer Output Analysis"
output: html_notebook
---

<!-- This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code.  -->

<!-- Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Cmd+Shift+Enter*.  -->

```{r setup, include=FALSE}
plot(cars)
```

<!-- Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Cmd+Option+I*. -->

<!-- When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Cmd+Shift+K* to preview the HTML file).  -->

<!-- The preview shows you a rendered HTML copy of the contents of the editor. Consequently, unlike *Knit*, *Preview* does not run any R code chunks. Instead, the output of the chunk when it was last run in the editor is displayed. -->

---

Can you pull out a couple examples of TFs (maybe 1 or 2) from these 73K and identify which part of the motif is affected by the SNP? I bet it will be the part of a motif where there is a very heavy weight on one particular nucleotide and messing that up results in failure to identify the motif with the alternative allele. 

also, could you show one or two examples of TFs from the 69K set, where searching the reference genome fails to identify a motif but searching with the alternative allele finds it? 

Similarly, I bet the SNPs occur in the super important core part of the motif with high probabilities (from the PWM) for that nucleotide. Do you have experience making sequence logos? theyâ€™re fun plots to show the important of each nucleotide in a motif?

Limitations: only looks at positive strand

```{r load_libraries}
# source("https://bioconductor.org/biocLite.R")
# BiocManager::install("seqLogo")

library(seqLogo) # makePWM
library(stringr) # read.fasta, write.fasta
library(data.table)
library(dplyr)
library(glue)
library(matrixStats)
```

```{r load_data}

in_ref_not_alt_unique_SNP <- fread("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_output/in_ref_not_alt_unique_SNP.txt") # 264905 x 8
in_alt_not_ref_unique_SNP <- fread("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_output/in_alt_not_ref_unique_SNP.txt") # 269853 x 8

x1 <- in_ref_not_alt_unique_SNP[!duplicated(in_ref_not_alt_unique_SNP),] # 70550 x 8
x2 <- in_alt_not_ref_unique_SNP[!duplicated(in_alt_not_ref_unique_SNP),] # 72078 x 8
#in_ref_not_alt_unique_SNP$ref_Offset <- in_ref_not_alt_unique_SNP$ref_Offset + 21
#in_alt_not_ref_unique_SNP$alt_Offset <- in_alt_not_ref_unique_SNP$alt_Offset + 21

a_gene_snp <- fread("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_output/a_gene_snp.txt") # 925540 x 14
a_gene_snp$SNPWindow <- paste("chromosome:GRCh37:",a_gene_snp$SNPChr, ":", a_gene_snp$SNPWindowStart, ":", a_gene_snp$SNPWindowEnd,":1", sep="") # 925540 x 15

null_ref <- fread("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_input/homer_master_ref.txt", sep='\t', header=T)
colnames(null_ref) <- c("SNPWindow", "ref_Offset", "Sequence", "Motif Name", "ref_Strand", "ref_MotifScore")
null_ref <- null_ref[!duplicated(null_ref),]
null_ref$SNPWindow <- gsub("\\-.*", "", null_ref$SNPWindow) # if necessary
null_alt <- fread("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_input/homer_master_alt.txt", sep='\t', header=T)
colnames(null_alt) <- c("SNPWindow", "alt_Offset", "Sequence", "Motif Name", "alt_Strand", "alt_MotifScore")
null_alt <- null_alt[!duplicated(null_alt),]
null_alt$SNPWindow <- gsub("\\-.*", "", null_alt$SNPWindow) # if necessary

x1 <- null_ref
x2 <- null_alt

```

Using only motifs on positive strands, sort motifs by Motif Score
Change # data points with value of n in z_test_group
```{r get_samples}

x1_s <- x1[!duplicated(x1$"Motif Name"), ] # 423 x8
x2_s <- x2[!duplicated(x2$"Motif Name"), ] # 422 x 8

# z <- inner_join(x1_s[x1_s$"ref_Strand" == "+",], x2_s[x2_s$"alt_Strand" == "+",], by="Motif Name") # 925540 x 9

z <- x1_s[x1_s$"ref_Strand" == "+",]
z_test_group_ref <- z[order(z$"ref_MotifScore", decreasing=F),]
# z_test_group_ref <- tail(z[order(z$"ref_MotifScore", decreasing=F),], n=126L) # 237 x 15
#z_test_group$alt_Offset <- z_test_group$alt_Offset + 21
#z_test_group$ref_Offset <- z_test_group$ref_Offset + 21

z <- x2_s[x2_s$"alt_Strand" == "+",]
z_test_group_alt <- z[order(z$"alt_MotifScore", decreasing=F),]
# z_test_group_alt <- tail(z[order(z$"alt_MotifScore", decreasing=F),], n=215L) # 215 x 15

```

http://zevross.com/blog/2014/08/05/using-the-r-function-anti_join-to-find-unmatched-records/
```{r understanding_anti_join, include=F}
table1[!table1$id%in%table2$id,]
table1[is.na(match(table1$id,table2$id)),]
anti_join(table1, table2, by="id")
```

```{r display_data, include=F}

ref_test_group_transposed <- t(x1_s[x1_s$"Motif Name" %in% z_test_group$"Motif Name",c("SNPWindow", "ref_Offset", "Motif Name", "Sequence")])
alt_test_group_transposed <- t(x2_s[x2_s$"Motif Name" %in% z_test_group$"Motif Name",c("SNPWindow", "alt_Offset", "Motif Name", "Sequence")])

ref_test_group_transposed
alt_test_group_transposed
```

```{r test_group_data}
ref_test_group <- x1_s[x1_s$"Motif Name" %in% z_test_group_ref$"Motif Name"] # 237 x 8 
alt_test_group <- x2_s[x2_s$"Motif Name" %in% z_test_group_alt$"Motif Name"] # 215 x 8

# Note: ref_test_group[1,"Motif Name"] != ref_test_group$`Motif Name`[1]
```

Todo: detect # motifs without hardcoding it in

```{r create_motifs_table}

# used forklift to rename all the motifs

motifs_table <- data.frame(read.table("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_investigation/motifs/motif1.motif", sep="\t", header=F,nrow=1)[,2])
colnames(motifs_table) <- "Motif Name"

n = 426
for (i in 2:n){
  z <- read.table(paste("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_investigation/motifs/motif", i, ".motif", sep=""), sep="\t", header=F,nrow=1)[,2]
  z <- data.frame(z)
  colnames(z) <- "Motif Name"
  motifs_table <- rbind(motifs_table, z)
}

motifs_table <- cbind(motifs_table, 1:n)
colnames(motifs_table)[2] <- "Num"
# 423 x 2

```

```{r match_motifs_with_test_groups_ref}
 
ref_test_group_file <- left_join(ref_test_group, motifs_table) # 11 x 9
ref_test_group_file$ref_Offset <- ref_test_group_file$ref_Offset + 21
ref_test_group_file$SNPPosInFasta <- 0.001 

ref_test_group_with_fasta <- merge(ref_test_group, a_gene_snp) # also by 'SNP'
ref_test_group_with_fasta$ref_Offset <- ref_test_group_with_fasta$ref_Offset+21
ref_test_group_with_fasta <- ref_test_group_with_fasta[which(str_sub(ref_test_group_with_fasta$fasta_seq_og, 21, 21) == ref_test_group_with_fasta$OtherAllele), ] # 37x20, only necessary if you didn't use old a_gene_snp file for homer input
counter_of_excluded_data <- 0
```

```{r get_max_pwm_info_ref}

i=1

## make_PWM_and_get_motif_sequence_for_1_motif
impt_table <- read.table(paste("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_investigation/motifs/motif", ref_test_group_file[i,"Num"], ".motif", sep=""), sep="\t", skip=1, header=F, stringsAsFactors=FALSE)
colnames(impt_table) <- c("A", "C", "G", "T")
max_TF_motifs <- data.frame(cbind(data.frame(rowMaxs(as.matrix(impt_table), cols=T)), 1:nrow(impt_table), colnames(impt_table)[apply(impt_table,1,which.max)]))
colnames(max_TF_motifs) <- c("Num", "Index", "Base")
tf_motif <- paste(max_TF_motifs$Base, collapse="", sep="")

## get_fasta_seq_and_get_rid_of_matches_that_do_not_work: Where does the SNP hit the TF? for alt_test_group: I think eQTLGen is zero-indexed.

## see_where_TF_hits_fasta_seq
motif_w_fasta_seq_info <- ref_test_group_with_fasta[which(ref_test_group_with_fasta$"Motif Name" == ref_test_group_file[i,"Motif Name"]),]
motif_w_fasta_seq_info <- motif_w_fasta_seq_info[1,] # only need first row for fasta seq

## does_TF_hit_SNP_if_so_get_position

if (motif_w_fasta_seq_info$ref_Offset[i] >= (21 - nchar(tf_motif)) & motif_w_fasta_seq_info$ref_Offset != 21)
{
    ref_test_group_file$SNPPosInFasta[i] <- 21 - ref_test_group_file$ref_Offset[i]
    all_snp_pwms_ref <- max_TF_motifs[21 - motif_w_fasta_seq_info$ref_Offset[i], ] # make this rbind later
    all_snp_pwms_ref <- cbind(all_snp_pwms_ref, i) # make this cbind later
} else {
  all_snp_pwms_ref <- data.table(NA, NA, NA, i) # to begin
  colnames(all_snp_pwms_ref) <- c("Num", "Index", "Base", "i")
  counter_of_excluded_data <- counter_of_excluded_data + 1
}
    
for (i in 2:nrow(ref_test_group_file)){
  impt_table <- read.table(paste("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_investigation/motifs/motif", ref_test_group_file[i,"Num"], ".motif", sep=""), sep="\t", skip=1, header=F, stringsAsFactors=FALSE)
  colnames(impt_table) <- c("A", "C", "G", "T")
  max_TF_motifs <- data.frame(cbind(data.frame(rowMaxs(as.matrix(impt_table), cols=T)), 1:nrow(impt_table), colnames(impt_table)[apply(impt_table,1,which.max)]))
  colnames(max_TF_motifs) <- c("Num", "Index", "Base")
  tf_motif <- paste(max_TF_motifs$Base, collapse="", sep="")  # CATGCATATGCATAT
  
  ## get_fasta_seq_and_get_rid_of_matches_that_do_not_work: Where does the SNP hit the TF? for ref_test_group: I think eQTLGen is zero-indexed.
  
  ## see_where_TF_hits_fasta_seq
  motif_w_fasta_seq_info <- ref_test_group_file[which(ref_test_group_file$"Motif Name" == ref_test_group_file[i,"Motif Name"]),]
  motif_w_fasta_seq_info <- motif_w_fasta_seq_info[1,] # only need first row for fasta seq
  
  ## does_TF_hit_SNP_if_so_get_position
  # plot(max_TF_motifs$Num)
  if (motif_w_fasta_seq_info$ref_Offset[1] > (21 - nchar(tf_motif)) & motif_w_fasta_seq_info$ref_Offset != 21) # if offset = 21, skip this
  {
      ref_test_group_file$SNPPosInFasta[i] <- 21 - motif_w_fasta_seq_info$ref_Offset[1]
      row_to_bind <- data.frame(max_TF_motifs[21 - motif_w_fasta_seq_info$ref_Offset[1], ], i)
      all_snp_pwms_ref <- rbind(all_snp_pwms_ref, row_to_bind)
  } else {
    counter_of_excluded_data <- counter_of_excluded_data + 1
  }
}

all_snp_pwms_ref <- all_snp_pwms_ref[!is.na(all_snp_pwms_ref$Num),]
  
```

```{r visualize_likelihood_ref}

# hist(all_snp_pwms$Num) # all_snp_pwms %>% count(Num)

# mean = blue dashed line, density plot = red transparent

ggplot(all_snp_pwms_ref, aes(x=as.numeric(Num))) + 
  #stat_count() +
  geom_histogram(binwidth=.1, color="black", fill="blue") +
  #y_lim(0, 4)
  #xlim(0, 1)+
  #geom_text(aes(label=Base), hjust=2, vjust=1) + 
  #geom_point(data = max_TF_motifs[21 - ref_test_group_with_fasta$ref_Offset[1],], x=Index, y = Num) +
  theme_stata() +
  geom_density(alpha=.2, fill="#FF6666", aes(y=..count../8)) +
  geom_vline(aes(xintercept=mean(as.numeric(Num))),
            color="blue", linetype="dashed", size=1) + 
  ggtitle("Likelihood of All Motifs - Reference Allele")

```

For alt

```{r match_motifs_with_test_groups_alt}

alt_test_group_file <- left_join(alt_test_group, motifs_table) # 11 x 9
alt_test_group_file$alt_Offset <- alt_test_group_file$alt_Offset + 21
alt_test_group_file$SNPPosInFasta <- 0.001 

alt_test_group_with_fasta <- merge(alt_test_group, a_gene_snp) # also by 'SNP'
alt_test_group_with_fasta$alt_Offset <- alt_test_group_with_fasta$alt_Offset+21
alt_test_group_with_fasta <- alt_test_group_with_fasta[which(str_sub(alt_test_group_with_fasta$fasta_seq_og, 21, 21) == alt_test_group_with_fasta$OtherAllele), ] # 37x20, only necessary if you didn't use old a_gene_snp file for homer input
counter_of_excluded_data <- 0

```

```{r get_max_pwm_info_alt}

i=1

## make_PWM_and_get_motif_sequence_for_1_motif
impt_table <- read.table(paste("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_investigation/motifs/motif", alt_test_group_file[i,"Num"], ".motif", sep=""), sep="\t", skip=1, header=F, stringsAsFactors=FALSE)
colnames(impt_table) <- c("A", "C", "G", "T")
max_TF_motifs <- data.frame(cbind(data.frame(rowMaxs(as.matrix(impt_table), cols=T)), 1:nrow(impt_table), colnames(impt_table)[apply(impt_table,1,which.max)]))
colnames(max_TF_motifs) <- c("Num", "Index", "Base")
tf_motif <- paste(max_TF_motifs$Base, collapse="", sep="")

## get_fasta_seq_and_get_rid_of_matches_that_do_not_work: Where does the SNP hit the TF? for alt_test_group: I think eQTLGen is zero-indexed.

## see_where_TF_hits_fasta_seq
motif_w_fasta_seq_info <- alt_test_group_with_fasta[which(alt_test_group_with_fasta$"Motif Name" == alt_test_group_file[i,"Motif Name"]),]
motif_w_fasta_seq_info <- motif_w_fasta_seq_info[1,] # only need first row for fasta seq

## does_TF_hit_SNP_if_so_get_position

if (motif_w_fasta_seq_info$alt_Offset[i] >= (21 - nchar(tf_motif)) & motif_w_fasta_seq_info$alt_Offset != 21)
{
    alt_test_group_file$SNPPosInFasta[i] <- 21 - alt_test_group_file$alt_Offset[i]
    all_snp_pwms_alt <- max_TF_motifs[21 - motif_w_fasta_seq_info$alt_Offset[i], ] # make this rbind later
    all_snp_pwms_alt <- cbind(all_snp_pwms_alt, i) # make this cbind later
} else {
  all_snp_pwms_alt <- data.table(NA, NA, NA, i) # to begin
  colnames(all_snp_pwms_alt) <- c("Num", "Index", "Base", "i")
  counter_of_excluded_data <- counter_of_excluded_data + 1
}
    
for (i in 2:nrow(alt_test_group_file)){
  impt_table <- read.table(paste("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_investigation/motifs/motif", alt_test_group_file[i,"Num"], ".motif", sep=""), sep="\t", skip=1, header=F, stringsAsFactors=FALSE)
  colnames(impt_table) <- c("A", "C", "G", "T")
  max_TF_motifs <- data.frame(cbind(data.frame(rowMaxs(as.matrix(impt_table), cols=T)), 1:nrow(impt_table), colnames(impt_table)[apply(impt_table,1,which.max)]))
  colnames(max_TF_motifs) <- c("Num", "Index", "Base")
  tf_motif <- paste(max_TF_motifs$Base, collapse="", sep="")  # CATGCATATGCATAT
  
  ## get_fasta_seq_and_get_rid_of_matches_that_do_not_work: Where does the SNP hit the TF? for alt_test_group: I think eQTLGen is zero-indexed.
  
  ## see_where_TF_hits_fasta_seq
  motif_w_fasta_seq_info <- alt_test_group_file[which(alt_test_group_file$"Motif Name" == alt_test_group_file[i,"Motif Name"]),]
  motif_w_fasta_seq_info <- motif_w_fasta_seq_info[1,] # only need first row for fasta seq
  
  ## does_TF_hit_SNP_if_so_get_position
  # plot(max_TF_motifs$Num)
  if (motif_w_fasta_seq_info$alt_Offset[1] > (21 - nchar(tf_motif)) & motif_w_fasta_seq_info$alt_Offset != 21) # if offset = 21, skip this
  {
      alt_test_group_file$SNPPosInFasta[i] <- 21 - motif_w_fasta_seq_info$alt_Offset[1]
      row_to_bind <- data.frame(max_TF_motifs[21 - motif_w_fasta_seq_info$alt_Offset[1], ], i)
      all_snp_pwms_alt <- rbind(all_snp_pwms_alt, row_to_bind)
  } else {
    counter_of_excluded_data <- counter_of_excluded_data + 1
  }
}

all_snp_pwms_alt <- all_snp_pwms_alt[!is.na(all_snp_pwms_alt$Num),]
```


```{r make_seqLogos, include=F}

pe <- makePWM(t(read.table(paste("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_investigation/motifs/motif", alt_test_group_file[i,"Num"], ".motif", sep=""), sep="\t", skip=1, header=F, stringsAsFactors=FALSE)))
pdf("Pax7_longest_bitsscore.pdf")
seqLogo(pe,ic.scale=T)
dev.off()

pe <- makePWM(t(foxa1))
pdf("Foxa1_lncap.bitsscore.pdf")
seqLogo(pe,ic.scale=T)
dev.off()
# AAAGTAAACA

pe <- makePWM(t(ronin))
pdf("Ronin_bitsscore.pdf")
seqLogo(pe,ic.scale=T)
dev.off()

```

```{r compare_TFs_and_sequence, include=F}

ref <- (x1_s[x1_s$"Motif Name"==ref_test_group_file[i,"Motif Name"], c("SNPWindow", "ref_Offset", "Motif Name", "ref_Strand", "Sequence")])

```

```{r, include=F}

ref_with_fasta <- merge(ref_pax7, a_gene_snp, by = c("SNPWindow"))

alt_pax7 <- (x2_s[x2_s$"Motif Name"=="Pax7(Paired,Homeobox),longest/Myoblast-Pax7-ChIP-Seq(GSE25064)/Homer", c("SNPWindow", "alt_Offset", "Motif Name", "alt_Strand", "Sequence")])

ref_ronin <- (x1_s[x1_s$"Motif Name"=="Ronin(THAP)/ES-Thap11-ChIP-Seq(GSE51522)/Homer", c("SNPWindow", "ref_Offset", "Motif Name", "ref_Strand", "Sequence")])
alt_ronin <- (x2_s[x2_s$"Motif Name"=="Ronin(THAP)/ES-Thap11-ChIP-Seq(GSE51522)/Homer", c("SNPWindow", "alt_Offset", "Motif Name", "alt_Strand", "Sequence")])

ref_foxa1 <- (x1_s[x1_s$"Motif Name"=="FOXA1:AR(Forkhead,NR)/LNCAP-AR-ChIP-Seq(GSE27824)/Homer", c("SNPWindow", "ref_Offset", "Motif Name", "ref_Strand", "Sequence")])
alt_foxa1 <- (x2_s[x2_s$"Motif Name"=="FOXA1:AR(Forkhead,NR)/LNCAP-AR-ChIP-Seq(GSE27824)/Homer", c("SNPWindow", "alt_Offset", "Motif Name", "alt_Strand", "Sequence")])

a_gene_snp$SNPWindow <- paste("chromosome:GRCh37:",a_gene_snp$SNPChr, ":", a_gene_snp$SNPWindowStart, ":", a_gene_snp$SNPWindowEnd,":1", sep="")

ref_pax7 <- merge(ref_pax7, a_gene_snp, by = c("SNPWindow"))
alt_pax7 <- merge(alt_pax7, a_gene_snp, by = c("SNPWindow"))
ref_ronin <- merge(ref_ronin, a_gene_snp, by = c("SNPWindow"))
alt_ronin <- merge(alt_ronin, a_gene_snp, by = c("SNPWindow"))
ref_foxa1 <- merge(ref_foxa1, a_gene_snp, by = c("SNPWindow"))
alt_foxa1 <- merge(alt_foxa1, a_gene_snp, by = c("SNPWindow"))
```

<!-- ggplot2: -->
<!-- https://www.google.com/search?ei=HCkaXa-VH8ev5wLY2ZToBQ&q=plot+r+label+points&oq=plot+r+label+points&gs_l=psy-ab.3..0j0i22i30l4.1980.5905..6176...0.0..0.397.1828.8j5j0j1......0....1..gws-wiz.......0i71j35i39j0i67j0i20i263.ZsFovZZdwdI -->
<!-- https://www.r-bloggers.com/adding-labels-to-points-in-a-scatter-plot-in-r/ -->
<!-- https://rpubs.com/RatherBit/188960 -->
<!-- https://www.google.com/search?q=ggplot2+scatter+plot+label+points&oq=ggplot2+scatter+plot+label+points&aqs=chrome.0.0j33l3.5449j0j1&sourceid=chrome&ie=UTF-8 -->
<!-- https://stackoverflow.com/questions/15624656/label-points-in-geom-point -->
<!-- http://www.sthda.com/english/wiki/ggplot2-themes-and-background-colors-the-3-elements -->
<!-- https://www.google.com/search?q=create+line+graph+in+ggplot2&oq=create+line+graph+in+ggplot2&aqs=chrome..69i57j0l2.4168j0j1&sourceid=chrome&ie=UTF-8 -->
<!-- http://www.sthda.com/english/wiki/ggplot2-line-plot-quick-start-guide-r-software-and-data-visualization -->
<!-- https://www.google.com/search?q=change+color+of+one+dot+r&oq=change+color+of+one+dot+r&aqs=chrome..69i57j33l5.2916j0j1&sourceid=chrome&ie=UTF-8 -->
<!-- https://www.google.com/search?ei=jkwaXeOhCKfG5gKP9qyABA&q=change+color+of+one+dot+r+ggplot&oq=change+color+of+one+dot+r+ggplot&gs_l=psy-ab.3..33i299j33i160.760.1518..1588...0.0..0.388.1318.2-1j3......0....1..gws-wiz.......0i71.VWvkbLhqHNE -->

```{r visualize_likelihood_alt}

# hist(all_snp_pwms$Num) # all_snp_pwms %>% count(Num)

# mean = blue dashed line, density plot = red transparent

ggplot(all_snp_pwms_alt, aes(x=as.numeric(Num))) + 
  geom_histogram(binwidth=.1, color="black", fill="blue") +
  theme_stata() +
  geom_density(alpha=.2, fill="#FF6666", aes(y=..count../8)) +
  geom_vline(aes(xintercept=mean(as.numeric(Num))),
            color="blue", linetype="dashed", size=1) + 
  ggtitle("Likelihood of Interesting Motifs - Alternative Allele")

```

```{r all_null}
all_homer <- rbind(all_snp_pwms_alt, all_snp_pwms_ref)

ggplot(all_homer, aes(x=as.numeric(Num))) + 
  geom_histogram(binwidth=.1, color="black", fill="blue") +
  theme_stata() +
  geom_density(alpha=.2, fill="#FF6666", aes(y=..count../8)) +
  geom_vline(aes(xintercept=mean(as.numeric(Num))),
            color="blue", linetype="dashed", size=1) + 
  ggtitle("Likelihood of All Motifs - Ref/Alt Allele")

```


```{r plot_tf_data, include=F}
ggplot(max_TF_motifs, aes(x=Index, y=Num, label=Base)) + 
  geom_line() + 
  geom_point() +
  geom_text(aes(label=Base), hjust=2, vjust=1) + 
  #geom_point(data = max_TF_motifs[21 - ref_test_group_with_fasta$ref_Offset[1],], x=Index, y = Num) +
  theme_stata() +
  ggtitle("Homer TF Motif")
  # + theme(
  #   panel.background = element_rect(fill = "lightblue",
  #                                   colour = "lightblue",
  #                                   size = 0.5, 
  #                                   linetype = "solid"),
  #   panel.grid.major = element_line(size = 0.5, 
  #                                   linetype = 'solid',
  #                                   colour = "white"), 
  #   panel.grid.minor = element_line(size = 0.25, 
  #                                   linetype = 'solid',
  #                                   colour = "white")
  # )

```

```{r pipeline_backup, include=F}

## make_PWM_and_get_motif_sequence_for_1_motif
i=1
impt_table <- read.table(paste("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_investigation/motifs/motif", ref_test_group_file[i,"Num"], ".motif", sep=""), sep="\t", skip=1, header=F, stringsAsFactors=FALSE)
colnames(impt_table) <- c("A", "C", "G", "T")
# pe <- makePWM(t(impt))
max_TF_motifs <- data.frame(cbind(data.frame(rowMaxs(as.matrix(impt), cols=T)), 1:nrow(impt), colnames(impt)[apply(impt,1,which.max)]))
colnames(max_TF_motifs) <- c("Num", "Index", "Base")
tf_motif <- paste(max_TF_motifs$Base, collapse="", sep="")  # CATGCATATGCATAT

## get_fasta_seq_and_get_rid_of_matches_that_do_not_work: Where does the SNP hit the TF? for ref_test_group: I think eQTLGen is zero-indexed.

## see_where_TF_hits_fasta_seq
motif_w_fasta_seq_info <- ref_test_group_with_fasta[which(ref_test_group_with_fasta$"Motif Name" == ref_test_group_file[i,"Motif Name"]),]
motif_w_fasta_seq_info <- motif_w_fasta_seq_info[1,] # only need first row for fasta seq

# substr(ref_test_group_with_fasta$fasta_seq_og[1], ref_test_group_with_fasta$ref_Offset[1]+1, ref_test_group_with_fasta$ref_Offset[1]+1)
# it works

## does_TF_hit_SNP_if_so_get_position
# plot(max_TF_motifs$Num)
if (motif_w_fasta_seq_info >= (21 - nchar(tf_motif)))
{
    ref_test_group_with_fasta$SNPPosInFasta[i] <- 21 - ref_test_group_with_fasta$ref_Offset[i]
    all_snp_pwms <- max_TF_motifs[21 - ref_test_group_with_fasta$ref_Offset[i], ] # make this rbind
}

# nchar("CATGTATACATATGT") = 15
# TF: substr("ACATGGCACATGTATACATATGTAACAAACCTGCACATTGTG", 9, 9+15-1) = (fasta_sequence_og, ref_offset+1, ref_offset+nchar(tf_motif)-1)
# SNP: substr("ACATGGCA*C*ATGTATACATA*T*GTAACAAACCTGCACATTGTG", 21, 21)
#                       9             21
# position of SNP in fasta sequence: 21
# if offset >= 21 - (nchar(tf_motif))
#   position of SNP in TF: 21 - 9 + 1 = 21 - (ref_Offset + 1) + 1 = 21 - ref_Offset = 21 - 8


for (i in 2:nrow(ref_test_group_file)){
  impt_row <- read.table(paste("/Users/kathryntsai/OneDrive\ -\ Villanova\ University/College/2018-2019/Summer\ 2019/TFs_eQTLs_Research/RProjects/Project2/q2_3_investigation/motifs/motif", ref_test_group_file[i ,"Num"], ".motif", sep=""), sep="\t", skip=1, header=F, stringsAsFactors=FALSE)
  impt_row <- data.table(impt_row)
  colnames(impt_row) <- c("A", "C", "G", "T")
  impt_table <- rbind(impt_table, impt_row)
}
```

```

