# How many SNP-gene associations exist in both single cell and bulk datasets? Of these overlapping associations, are they more or less cell type specific than the associations that can only be detected in the single cell data?
# Do we even need sc-eQTLs if we can use IMPACT to identify the cell type? 



# Since Pfizer eQTLs are found in whole blood, can we use IMPACT to infer cell-type-specificity? Since eQTLgen eQTLs are found in peripheral blood, can we use IMPACT to infer cell-type-specificity? 

# After learning from the annotation analyses above, can we predict if a given SNP will participate in a cis or trans eQTL? Consider 3 groups of SNPs: only cis, only trans, both cis and trans (with two separate genes, of course) 
# What are the models of these datasets? Do all include expression PCs? Traditionally, expression PCs donâ€™t take away signal from cis eQTLs but they might from trans eQTLs. Need to know data details, e.g. how many donors? How many samples? How many cells? 

# proportion of pbmcs are different cells